selfdrive/thermald/thermald.py

```diff
@@ -381,9 +380,7 @@ def thermald_thread() -> NoReturn:
 
     # Check if we need to shut down
     if power_monitor.should_shutdown(peripheralState, onroad_conditions["ignition"], in_car, off_ts, started_seen):
-      cloudlog.info(f"shutting device down, offroad since {off_ts}")
-      # TODO: add function for blocking cloudlog instead of sleep
-      time.sleep(10)
+      cloudlog.blocking_info(f"shutting device down, offroad since {off_ts}")
```

Comment:
Maybe you should add a timeout flag as well in case it gets stuck. And maybe overriding the info function instead with a blocking argument if possible.

Reply:
Added the timeout. Not sure about overriding since that would mean every call to cloudlog.info() would have to do an additional boolean check which doesn't really make sense when there is only one call to the blocking version

Reply:
Either way works, just a suggestion! I'd also add an argument with default for the timeout, so you can see that explicitly in thermald

Reply:
agreed